packages/director/src/execution/mongo/runs/run.model.ts

```diff
@@ -3,15 +3,74 @@ import { getMongoDB } from '@src/lib/mongo';
 import { AppError, RUN_EXISTS, CLAIM_FAILED } from '@src/lib/errors';
 import { getSanitizedMongoObject } from '@src/lib/results';
 
+const mergeRunSpecs = (run: any) => {
+  // merge fullspec into spec
+  run.specs = run.specs.map((spec: any) =>
+    Object.assign(
+      {},
+      spec,
+      run.specsFull.find((full: any) => full.instanceId === spec.instanceId) ||
+        {}
+    )
+  );
+  return run;
+};
+
+const matchRunAggregation = (runId: string) => ({
+  $match: {
+    runId,
+  },
+});
+
+const projectAggregation = {
+  $project: {
+    _id: 1,
+    runId: 1,
+    meta: 1,
+    specs: 1,
+    createdAt: 1,
+    specsFull: {
+      $map: {
+        input: '$specs',
+        as: 'spec',
+        in: '$$spec.instanceId',
+      },
+    },
+  },
+};
+
+const lookupAggregation = {
+  $lookup: {
+    from: 'instances',
+    localField: 'specsFull',
+    foreignField: 'instanceId',
+    as: 'specsFull',
+  },
+};
+
+export const getRunWithSpecs = async (id: string) =>
+  mergeRunSpecs(
+    (
+      await getMongoDB()
+        .collection('runs')
+        .aggregate([
+          matchRunAggregation(id),
+          projectAggregation,
+          lookupAggregation,
+        ])
+        .toArray()
+    ).pop()
+  );
+
 export const getRunById = async (id: string) =>
   await getMongoDB().collection('runs').findOne({ runId: id });
 
 export const createRun = async (run: Run) => {
   try {
-    const { result } = await getMongoDB()
+    const result = await getMongoDB()
       .collection('runs')
       .insertOne(getSanitizedMongoObject(run));
-    return result;
+    return result.ops[0];
```

Comment:
Why did it change? couldn't follow

Reply:
The original result is the mongo operation results. Which is not what I was expecting to get back when I told the execution driver to create a run. So I modified the result to return the actual created run from the mongo operation. Which in the case happend to be the result of the 0ith operation.

This is where the result is used.
https://github.com/sorry-cypress/sorry-cypress/pull/148/files#diff-642ad9f68f8d922e3064843951f75474R47